### PR TITLE
Increased length of clickable space in navbar

### LINKED ISSUE #279: Make links clickable screen-wide in collapsible menu
### What happened?

> the collapsible mobile menu should be using screen-wide buttons/links instead of a tiny segment of text

Reported by @diamondburned via Discord

### LINKED ISSUE #511: cannot see navbar separation on dark mode
### What happened?

image
image

### LINKED ISSUE #580: 1st: Confetti should show behind navbar
### What's your idea?

> Making the top header bar transparent for this particular page would be cool. It might be challenging to do this in Svelte with components.
> 
> image

Suggested by @diamondburned in <https://github.com/EthanThatOneKid/acmcsuf.com/pull/559#pullrequestreview-1114482570>.

### PR DESCRIPTION
**Pull request by k4rni**
Previously, in mobile mode, the size of the clickable area in the navbar was significantly small so it required you to press the names of the pages directly rather than the surrounding area itself.
Animation

Now it has been updated to encompass a larger area of clickable space. The size should be from one end of the screen to the other end.
Animation2

I am open to suggestions if whether we should have a title for the navbar, such as like "Menu", so that I can push down all the pages and make it easier to click if they are all situated near the bottom. I am also unsure of how to fix the words stacking on top of each other in "Node Buds". I am aware we have removed that page from the navbar, but for future pages it should not be stacked.

Fixes #279.

This PR additionally makes the navbar transparent, which addresses #580.

### CODE REVIEW

**Comment by k4rni** [comment]
@diamondburned This was an issue you pointed out so let me know what you think about this solution!

File: src/lib/components/sections/navbar.svelte [hunk 1]
```diff
@@ -205,6 +205,9 @@
 
           a {
             font-size: var(--size-md);
+            display: block;
+            padding: 30px 900px;
+            margin-top: -50px;
```
Comment (jaasonw) [inline src/lib/components/sections/navbar.svelte]: how come margin top here? make it asymmetrical i think
image

File: src/lib/components/sections/navbar.svelte [hunk 2]
```diff
@@ -205,6 +205,9 @@
 
           a {
             font-size: var(--size-md);
+            display: block;
+            padding: 30px 900px;
```
Comment (jaasonw) [inline src/lib/components/sections/navbar.svelte]: is possible to avoid hard coded values here or no?

**Review by jaasonw — CHANGES_REQUESTED** [review]

File: src/lib/components/sections/navbar.svelte (hunk 1 above)
Comment (k4rni) [reply src/lib/components/sections/navbar.svelte]: I shall try and fix this!

File: src/lib/components/sections/navbar.svelte (hunk 2 above)
Comment (k4rni) [reply src/lib/components/sections/navbar.svelte]: I will try I might be able to

Comment (diamondburned) [reply src/lib/components/sections/navbar.svelte]: A cheap (imo) workaround is to just turn the navbar into a sidebar when the screen is narrow. That way, it can sit on the right with a fixed width.

**Comment by k4rni** [comment]
Fixes #511 as well

**Review by diamondburned — APPROVED** [review]

**Comment by diamondburned** [comment]
I prefer the old dark navbar better with the dark theme, but I think the new changes are overall more consistent with the design.

**Comment by k4rni** [comment]
I'll try to actually change it back to the dark theme navbar but add perhaps a shadow.

File: src/lib/components/nav/bar.svelte [hunk 3]
```diff
@@ -193,16 +206,27 @@
           align-items: center;
           top: 0;
           right: 0;
-          height: 0;
-          width: 100%;
+          height: 100vh;
```
Comment (jaasonw) [inline src/lib/components/nav/bar.svelte]: remove, height 100% already inherit from line 138

File: src/lib/components/nav/bar.svelte [hunk 4]
```diff
@@ -123,17 +125,28 @@
...
+
+        .pages {
+          gap: 50px;
           list-style: none;
 
           a {
             text-decoration: none;
             transition: 0.25s ease-in-out;
+            padding-left: 8px;
+            padding-right: 8px;
```
Comment (jaasonw) [inline src/lib/components/nav/bar.svelte]: replace w padding: 8px 8px

File: src/lib/components/nav/bar.svelte [hunk 5]
```diff
@@ -193,16 +206,27 @@
...
           gap: 0;
           background-color: var(--acm-light);
           overflow: hidden;
           z-index: 9;
           transition: 0.25s ease-in-out;
 
           a {
-            font-size: var(--size-md);
+            justify-content: center;
+            align-items: center;
```
Comment (jaasonw) [inline src/lib/components/nav/bar.svelte]: remove, align center inherit from dekstop mode

File: src/lib/components/nav/bar.svelte [hunk 6]
```diff
@@ -193,16 +206,27 @@
...
+            font-size: var(--size-lg);
+            padding: 0 !important;
```
Comment (jaasonw) [inline src/lib/components/nav/bar.svelte]: why padding 0 here? make clickable area smaller, see screshot
image
image

File: src/lib/components/nav/bar.svelte [hunk 7]
```diff
@@ -193,16 +206,27 @@
...
+            width: 100%;
+            text-align: center;
+          }
+
+          li {
+            display: flex;
```
Comment (jaasonw) [inline src/lib/components/nav/bar.svelte]: remove, inherit from desktop

File: src/lib/components/nav/bar.svelte [hunk 8]
```diff
@@ -240,7 +264,22 @@
 
       & :checked ~ .container {
         .pages {
+          width: 60%;
```
Comment (jaasonw) [inline src/lib/components/nav/bar.svelte:295]: can add transition to the backdrop? think it look better than immediate turn background black

File: src/lib/components/nav/bar.svelte [hunk 9]
```diff
@@ -123,17 +125,28 @@
...
         gap: 64px;
+        height: 100%;
 
-        .pages {
+        .pages,
+        li,
+        a {
           display: flex;
-          gap: 64px;
+          align-items: center;
```
Comment (jaasonw) [inline src/lib/components/nav/bar.svelte:161]: add a justify content center here so can remove line 218 and 205

Comment (jaasonw) [reply src/lib/components/nav/bar.svelte:161]: the one in .pages and the one in a

File: src/lib/components/nav/bar.svelte [hunk 10]
```diff
@@ -193,16 +206,27 @@
           align-items: center;
           top: 0;
           right: 0;
-          height: 0;
-          width: 100%;
+          height: 100vh;
+          width: 0;
           gap: 0;
```
Comment (jaasonw) [inline src/lib/components/nav/bar.svelte]: remove, doesnt do anything

File: src/lib/components/nav/bar.svelte [hunk 11]
```diff
@@ -193,16 +206,27 @@
...
+            align-items: center;
+            font-size: var(--size-lg);
+            padding: 0 !important;
+            width: 100%;
+            text-align: center;
+          }
+
+          li {
+            display: flex;
+            width: 100%;
```
Comment (jaasonw) [inline src/lib/components/nav/bar.svelte]: do 
```
li, a {
  width: 100%
}
```
avoid have 2 100%s

File: src/lib/components/nav/bar.svelte [hunk 12]
```diff
@@ -240,7 +264,22 @@
...
         .pages {
+          width: 60%;
+          overflow: visible;
+          background-color: var(--acm-light);
+        }
+
+        .backdrop {
+          position: absolute;
+          top: 0;
+          bottom: 0;
```
Comment (jaasonw) [inline src/lib/components/nav/bar.svelte]: remove, only need top left

File: src/lib/components/nav/bar.svelte [hunk 13]
```diff
@@ -240,7 +264,22 @@
...
+          left: 0;
+          right: 0;
```
Comment (jaasonw) [inline src/lib/components/nav/bar.svelte]: remove, only need top left

File: src/lib/components/nav/bar.svelte [hunk 14]
```diff
@@ -240,7 +264,22 @@
...
+          width: 100vw;
           height: 100vh;
+          opacity: 0.7;
+          z-index: 4;
```
Comment (jaasonw) [inline src/lib/components/nav/bar.svelte]: remvoe, already on correct layer by defaukt

File: src/lib/components/nav/bar.svelte [hunk 15]
```diff
@@ -123,17 +125,28 @@
 
       section {
         display: flex;
+        justify-content: center;
```
Comment (jaasonw) [inline src/lib/components/nav/bar.svelte]: remove, nothing to justify center on this axis

**Review by jaasonw — CHANGES_REQUESTED** [review]
this navbar need refactor tbh

**Review by jaasonw — CHANGES_REQUESTED** [review]
1. acm icon is above overlay
2. backdrop animation is too slow
3. mobile navbar is too tall

https://user-images.githubusercontent.com/8981287/195045156-3b6ed7d3-3bac-4f1f-80a4-a1bdd9a974d9.mp4

https://user-images.githubusercontent.com/8981287/195072989-a7a1b6f4-6bf5-48c4-8050-bda22f409540.mp4

**Comment by jaasonw** [comment]
https://user-images.githubusercontent.com/8981287/195186144-d0b01556-c286-4b28-8cfc-0bda6d945851.mp4

possible to disable scrolling here too?

**Review by jaasonw — CHANGES_REQUESTED** [review]
button not seem to work here

https://user-images.githubusercontent.com/8981287/195186510-888afc26-8add-442c-afcf-dd2252c5fd41.mp4

**Review by jaasonw — CHANGES_REQUESTED** [review]
image

File: src/lib/components/nav/bar.svelte [hunk 16]
```diff
@@ -238,13 +284,28 @@
...
         .pages {
-          height: 100vh;
+          width: 60%;
+          overflow: visible;
+          background-color: var(--acm-light);
         }
       }
 
       & :checked ~ .menu {
+        :global(body) {
```
Comment (diamondburned) [inline src/lib/components/nav/bar.svelte]: Does this actually work?

Comment (k4rni) [reply src/lib/components/nav/bar.svelte]: OH NO I FORGOT TO REMOVE THAT SORRY

File: svelte.config.js [hunk 17]
```diff
@@ -10,17 +10,6 @@ const config = {
   kit: {
     adapter: adapter(),
   },
-
```
Comment (JustinStitt) [inline svelte.config.js:15]: would almost like this to be a separate commit because it seems loosely related.

**Review by JustinStitt — APPROVED** [review]
looks good to me.

your contribution improves the mobile experience.

we are, however, starting to realize the technical debt with the current navbar structure. Perhaps a refactor is needed.

we may also want to package up some of these frequently used css properties like:
```
position: absolute;
width: 100%;
height: 100%;
```
into their own class to reduce code bloat/smell.

we may also want to reduce the css nesting levels. i am not sure on the best approach but some of these elements are nested 5-6 levels deep and all within a media query as well. unmaintainable!!! (separate issue to be solved alongside the html refactor prob)

**Review by jaasonw — CHANGES_REQUESTED** [review]
1. logo z index wrong again
2. overlay transition seem to not correct

https://user-images.githubusercontent.com/8981287/195940787-bd0e3e81-6f0c-4174-a9d5-d656fee362e8.mp4

**Comment by k4rni** [comment]
When you get a chance, please take a look at this @jaasonw! Thank you!

**Comment by jaasonw** [comment]
the reason it's taking so long is cus im still looking into fixing the exit animation and havent had the time

**Comment by k4rni** [comment]
We can make it a separate issue if you would like, but if not I will also look into it as well. Thank you for letting me know.

**Review by jaasonw — APPROVED** [review]
fixed
